### PR TITLE
use cmake toolchain to configure build

### PR DESCRIPTION
**Pull request by mxyng**

### CODE REVIEW

File: Dockerfile.cuda [hunk 1]
```diff
@@ -1,8 +1,10 @@
 FROM nvidia/cuda:12.2.0-devel-ubuntu22.04
```
Comment (jmorganca) [inline Dockerfile.cuda:1]: Might need to set `GOARCH=$TARGETARCH` in the `go build` command below

**Comment by jmorganca** [comment]
Does cross compiling with `GOARCH` still work with this? Also, this doesn't seem to simplify things – we considered using cmake toolchain files previously but it's another thing to understand how the build works

File: Dockerfile.cuda (hunk 1 above)
Comment (jmorganca) [inline Dockerfile.cuda:1]: Exciting that there's `arm64` and `amd64` versions of this base image!!

File: llm/llama.cpp/generate_linux.go [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 //go:generate -command git-apply git -C ggml apply
 //go:generate git-apply ../ggml_patch/0001-add-detokenize-endpoint.patch
 //go:generate git-apply ../ggml_patch/0002-34B-model-support.patch
-//go:generate cmake -S ggml -B ggml/build/gpu -DLLAMA_CUBLAS=on -DLLAMA_ACCELERATE=on -DLLAMA_K_QUANTS=on 
+//go:generate cmake-toolchain -S ggml -B ggml/build/gpu
 //go:generate cmake --build ggml/build/gpu --target server --config Release
 
 //go:generate git submodule update --force gguf
-//go:generate cmake -S gguf -B gguf/build/gpu -DLLAMA_CUBLAS=on -DLLAMA_ACCELERATE=on -DLLAMA_K_QUANTS=on 
+//go:generate cmake-toolchain -S ggml -B ggml/build/gpu
```
Comment (BruceMacD) [inline llm/llama.cpp/generate_linux.go:15]: good catch

**Review by BruceMacD — APPROVED** [review]
looks good once Jeff's comments are addressed

**Comment by mxyng** [comment]
> Does cross compiling with GOARCH still work with this?

It turns out it doesn't until the second run so going to close this

**Pull request closed without merging**